ui/server/src/main/scala/pl/touk/nussknacker/ui/process/ProcessService.scala

```diff
@@ -252,6 +258,18 @@ class DBProcessService(managerActor: ActorRef,
       }).toSet)
   }
 
+  def importProcess(processId: ProcessIdWithName, jsonString: String)(implicit user: LoggedUser): Future[XError[ValidatedDisplayableProcess]] = {
+    withNotArchivedProcess(processId, "Import is not allowed for archived process.") { process =>
+      val result = for {
+        jsonCanonical <- ProcessMarshaller.fromJson(jsonString).leftMap(UnmarshallError).toEither
+        canonical = jsonCanonical.withProcessId(processId.name)
+        displayable = ProcessConverter.toDisplayable(canonical, process.processingType)
+      } yield new ValidatedDisplayableProcess(displayable, processValidation.validate(displayable))
```

Comment:
what happens if we use processResolving.validateBeforeUiResolving(...) - like in process update?

Reply:
Good catch, we can't remove processValidation and use processResolving instead of it.